packages/astro/src/content/utils.ts

```diff
@@ -167,6 +167,50 @@ export function getEntryConfigByExtMap<TEntryType extends ContentEntryType | Dat
 	return map;
 }
 
+export async function getSymlinkedContentCollections(
+	contentDir: URL
+): Promise<Map<string, string>> {
+	const contentPaths = new Map<string, string>();
+	const contentDirPath = fileURLToPath(contentDir);
+
+	if (!fsMod.existsSync(contentDirPath) || !fsMod.lstatSync(contentDirPath).isDirectory()) {
+		return contentPaths;
+	}
+
+	const contentDirEntries = await fsMod.promises.readdir(contentDir, { withFileTypes: true });
+	for (const entry of contentDirEntries) {
+		if (entry.isSymbolicLink()) {
+			const entryPath = path.join(contentDirPath, entry.name);
+			const realPath = await fsMod.promises.realpath(entryPath);
+			contentPaths.set(normalizePath(realPath), entry.name);
+		}
+	}
+	return contentPaths;
+}
+
+export function reverseSymlinks({
```

Comment:
nit: `reverseSymlinks` -> `reverseSymlink` singular, because it returns a `string`, and not a `stringp[]`

Reply:
The reason I did the plural is because there are multiple symlinks. But I can see what you mean.